### ISSUE TITLE
Feature request: Collabora Online for shared hosting

### ISSUE THREAD

**Issue by sushidave** [nextcloud/richdocuments#29]
While only few people out there have a server with root access many use Nextcloud on a cheaper shared hosting but they can't use Collabora Online, right? How about enhancing it to be used on a shared hosting, like Nextcloud?

**Comment by basille** [comment]
I second @sushidave here: are there any plans to provide Collabora Online (or any other office suite, really) for NextCloud on shared hosting? Nobody commented on this issue, it would be nice to have an "official" NextCloud feedback on it: Is this feasible? Planned? Not an option? As far as I can tell, there is currently no solution for an office suite with NextCloud on shared hosting.

(it may be what @eranj is hoping for in #116)

**Comment by timar** [comment]
What is missing from Collabora Online that specifically required with shared hosting? Maybe I wrong, but I think it's something to discuss with the hoster. I'm open to suggestions.

**Comment by PetrusVermaak** [comment]
@timar Well first of all no sudo, and I cannot configure the apache server or change anything. I can however install nextcloud via the web installer just fine!. I can also upload the nextcloud zip file, unpack and configure. Would be awesome if Collabora Online can do the same!!

**Comment by ectatomma** [comment]
@timar  I just tried to negotiate with my hosting provider and they said it's not  possible in a shared hosting, I need to pay for a VPS or a dedicated server. I know Docker is a super easy way to distribute a preconfigured server, but would be great to have an alternative compatible with the restrictions of a shared hosting, like @PetrusVermaak said.

**Comment by Gajusbonus** [comment]
Could undocker help here?
http://blog.oddbit.com/2015/02/13/unpacking-docker-images/

**Comment by eranj** [comment]
@Gajusbonus No. It needs to be similar to Nextcloud installation for shared hosting. PHP files, MySQL database with images, css etc.

**Comment by sushidave** [comment]
@rullzer Thanks for your comment.

Yes, it currently fails due to permissions. This feature request was about the idea of enhancing Collabora for shared hosting. What if it could be turned into a dockerless Nextcloud app that can be easily installed and activated ....or an embedded commercial Saas thing provided by Collabora or by Nextcloud GmbH? Well, I'm dreaming. Let's see if there will be a collaborative office suite for Nextcloud for shared hosting in the future. After all, Nexcloud used to be a dream too :-)

**Comment by PetrusVermaak** [comment]
Still hoping to have this some day... Nextcloud works great without sudo...

**Comment by Kogoro** [comment]
Would also like to see a change on this topic. It is the only showstopper for me when using shared hosting.

**Comment by jorsn** [comment]
> If you have shared hosting I really don't think your hoster will allow you to just run arbitrary processes on them.
> 
> Collabora is not just php+mysql etc. So it won't work that way.

@rullzer  **You might have to rethink:** With my hoster, [uberspace](https://uberspace.de), I own a normal linux user account with ssh login and can do about everything you can do as such. You also have a public content/cgi-bin/fcgi-bin directory (and possibly more for subdomains) and some preconfigured services/preinstalled software like dbs, web server, git, etc, email servers, etc. Additionally you can open ports for your own services and configure a reverse proxy to access them easily.

**Restrictions** are mostly:

* no root access
* no docker, because you (usually; so far) need to be approximately root to control the docker daemon
* limited disk, memory and cpu resources

---
Bottomline: There are shared hosters where an installation of someting like CODE should be possible. According to my research/experiments, the reason this is not possible is CODE's requirement of root permissions, which I found non-trivial to remove (1 or 2 years ago). This is not in the control domain of Nextcloud. **However, if this is a use case specifically relevant to Nextcloud users, Nextcloud could propose corresponding changes to Collabora.**

As a workaround, manual user+mount-namespace magic [proot](https://proot-me.github.io/) might help.
